### ISSUE TITLE
PDFViewerApplication annotations misplaced

### ISSUE THREAD

**Issue by IrvinDominin** [mozilla/pdf.js#7434]
Link to PDF file (or attach file here): http://79.99.8.89/test/web/compressed.tracemonkey-pldi-09.pdf

Configuration:

Web browser and its version: Android 5.01 webview
Operating system and its version: Samsung Galaxy Tab with Android 5.0.1
PDF.js version: 1.3.91

In order to use less memory possibile I'm loading the file in chunks and here is the configuration code:

```
PDFJS.disableWorker = true;
PDFJS.maxCanvasPixels = 393216;
```

I'm using pdfviewer in my application; the viewer works fine and is loaded ad documented here [https://github.com/mozilla/pdf.js/issues/7127](url)

The pdf have many data annotation; very often in the scenario presented in configuration the annotations are misplaced.

Often (not always) I see the issue in this way:
1. scroll to page 12
2. scroll a lot of pages and go to page 90
3. using the page selector, go to page 12

Now I see the data annotation misplaced like:
![image](https://cloud.githubusercontent.com/assets/1533624/16338755/f0d36ee6-3a1e-11e6-9bce-7284013c097a.png)

but it must be:
![image](https://cloud.githubusercontent.com/assets/1533624/16338765/fa58fd14-3a1e-11e6-9d07-f0c4bca2c1ce.png)

How can I fix the issue? I'm trying a lot, but I'm unable to reproduce this issue in a desktop browser...

Kind regards,
Irvin Dominin

**Comment by IrvinDominin** [comment]
Hi @timvandermeij do you have any idea on this? What can I check?

**Comment by timvandermeij** [comment]
Try using the most recent PDF.js version (https://github.com/mozilla/pdf.js/releases). I recall there was an issue about flipped annotations before that has been fixed. The link to the PDF file above is dead, so please update it as well (or better, upload to GitHub).

**Comment by IrvinDominin** [comment]
Hi @timvandermeij thanks; I try with the latest version, I update you and eventually close this issue ASAP. Here is the file [Catalogo_XS_1.pdf](https://github.com/mozilla/pdf.js/files/645335/Catalogo_XS_1.pdf)

**Comment by Snuffleupagus** [comment]
> I recall there was an issue about flipped annotations before that has been fixed.

Are you perhaps thinking of PR #6814, by any chance? If so, I'm unfortunately not sure if that will help here, since the annotations that this issue is about appear to just be standard `LinkAnnotations` with a border.

---

> The pdf have many data annotation; very often in the scenario presented in configuration the annotations are misplaced.

The "very often" part seems to suggest that this is an intermittent issue, thus making any sort of debugging quite hard.

> Web browser and its version: Android 5.01 webview
> Operating system and its version: Samsung Galaxy Tab with Android 5.0.1
> [...]
> I'm trying a lot, but I'm unable to reproduce this issue in a desktop browser...

This further suggests that the issue might be platform and/or even browser specific, thus making debugging *even* harder.

So, to summarize: Based on all of the information provided, it seems very likely that this is a browser (or platform) specific bug, and thus unfortunately most likely not something that can be fixed on the PDF.js side.

**Comment by Snuffleupagus** [comment]
Closing, since based on https://github.com/mozilla/pdf.js/issues/7434#issuecomment-266419093 there doesn't appear to be much that can really be done about this in PDF.js unfortunately.
